envoimoinscher.php

```diff
@@ -3946,6 +4024,299 @@ private function postProcess()
 
 	private function postProcessIntroduction()
 	{
+		Configuration::updateValue('EMC_USER', -1);
+	}
+
+	private function postProcessEmc()
+	{
+		$errors = array();
+		$helper = new EnvoimoinscherHelper();
+
+		// Validation for account creation form
+		if (Tools::getValue('choice') == 'create')
+		{
+
+			// validate gender
+			if (Tools::getValue('contact_civ'))
+			{
+				if (Tools::getValue('contact_civ') == 'M.')
+					Configuration::updateValue('EMC_CIV', 'M');
+				else
+					Configuration::updateValue('EMC_CIV', 'Mme');
+			}
+
+			// validate surname
+			if (Tools::getValue('contact_nom'))
+				Configuration::updateValue('EMC_LNAME', Tools::getValue('contact_nom'));
+			else
+				array_push($errors, $this->l('Please specify your surname'));
```

Comment:
http://php.net/manual/en/function.array-push.php

```Note: If you use array_push() to add one element to the array it's better to use $array[] = because in that way there is no overhead of calling a function. ```